crates/chain/src/tx_graph.rs

```diff
@@ -681,11 +705,29 @@ impl<A: Anchor> TxGraph<A> {
             }
         };
 
+        // Traverse tx and its unconfirmed ancestor txs to check for conflicts
+        let ancestor_txs =
+            TxAncestors::new_include_root(self, tx, |_, ancestor_tx: &Transaction| {
+                let tx_node = self.get_tx_node(ancestor_tx.txid())?;
+                for block in tx_node.anchors {
+                    match chain.is_block_in_chain(block.anchor_block(), chain_tip) {
+                        Ok(Some(true)) => return None,
+                        Err(e) => return Some(Err(e)),
+                        _ => continue,
+                    }
+                }
+                Some(Ok(tx_node.tx))
+            })
+            .collect::<Result<Vec<_>, C::Error>>()?;
+
         // If a conflicting tx is in the best chain, or has `last_seen` higher than this tx, then
         // this tx cannot exist in the best chain
-        for conflicting_tx in self.walk_conflicts(tx, |_, txid| self.get_tx_node(txid)) {
-            for block in conflicting_tx.anchors.iter().map(A::anchor_block) {
-                if chain.is_block_in_chain(block, chain_tip)? == Some(true) {
+        for conflicting_tx in ancestor_txs
+            .iter()
+            .flat_map(|&tx| self.walk_conflicts(tx, |_, txid| self.get_tx_node(txid)))
+        {
```

Comment:
Really small nit: for clarity, I would declare the iterator as a variable. Also, we can use `into_iter` to consume `ancestor_txs` since we aren't using it anymore.
```suggestion
        let conflicting_txs = ancestor_txs
            .into_iter()
            .flat_map(|ancestor| self.walk_conflicts(ancestor, |_, txid| self.get_tx_node(txid)));

        for conflicting_tx in conflicting_txs {
```